test/qd8-f16-qb4w-gemm-minmax.cc

```diff
@@ -0,0 +1,494 @@
+        "small_kernel_subtile",
+        GemmMicrokernelTester()
+            .mr(mr).nr(nr).kr(kr).sr(sr).ks(3).iterations(1)
+            .b_zero_point(8)
+            .bl(kr * sr * 2)
+        , test_func, isa_check)
+        .loop_k(1, k_block * 3, k_block + 1)
+        .loop_n(1, nr)
+        .loop_m(1, mr));
+    gemm_tests.push_back(GemmTestParams(
+        "n_gt_" + nrs + "_small_kernel",
+        GemmMicrokernelTester()
+            .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).ks(3)
+            .b_zero_point(8)
+            .bl(kr * sr * 2)
+        , test_func, isa_check)
+        .loop_n(nr + 1, nr * 2 - 1)
+        .loop_k(1, k_block * 3, k_block + 1));
+    gemm_tests.push_back(GemmTestParams(
+        "n_div_" + nrs + "_small_kernel",
+        GemmMicrokernelTester()
+            .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).ks(3)
+            .b_zero_point(8)
+            .bl(kr * sr * 2)
+        , test_func, isa_check)
+        .loop_n(nr * 2, nr * 3, nr)
+        .loop_k(1, k_block * 3, k_block + 1));
+  }
+  gemm_tests.push_back(GemmTestParams(
+      "strided_cm_subtile",
+      GemmMicrokernelTester()
+          .mr(mr).nr(nr).kr(kr).sr(sr)
+          .cm_stride(NextPrime(nr + 1))
+          .iterations(1)
+          .b_zero_point(8)
+          .bl(kr * sr * 2)
+      , test_func, isa_check)
+      .loop_k(1, k_block * 3, k_block + 1)
+      .loop_n(1, nr)
+      .loop_m(1, mr));
+  if (is_igemm) {
+    gemm_tests.push_back(GemmTestParams(
+        "a_offset",
+        GemmMicrokernelTester()
+            .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).ks(3)
+            .a_offset(NextPrime(mr * k_block * 3 + 1))
+            .b_zero_point(8)
+            .bl(kr * sr * 2)
+        , test_func, isa_check)
+        .loop_k(1, k_block * 3, k_block + 1));
+    gemm_tests.push_back(GemmTestParams(
+        "zero",
+        GemmMicrokernelTester()
+            .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).ks(3)
+            .a_offset(NextPrime(mr * k_block * 3 + 1))
+            .b_zero_point(8)
+            .bl(kr * sr * 2)
+        , test_func, isa_check)
+        .loop_k(1, k_block * 3, k_block + 1)
+        .loop_zi(0, mr - 1));
+  }
+  gemm_tests.push_back(GemmTestParams(
+      "qmin",
+      GemmMicrokernelTester()
+          .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).k(k_block).qmin(128)
+          .b_zero_point(8)
+          .bl(kr * sr * 2)
+      , test_func, isa_check));
+  gemm_tests.push_back(GemmTestParams(
+      "qmax",
+      GemmMicrokernelTester()
+          .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).k(k_block).qmax(128)
+          .b_zero_point(8)
+          .bl(kr * sr * 2)
+      , test_func, isa_check));
+  gemm_tests.push_back(GemmTestParams(
+      "strided_cm",
+      GemmMicrokernelTester()
+          .mr(mr).nr(nr).kr(kr).sr(sr).m(mr).n(nr).k(k_block)
+          .cm_stride(NextPrime(nr + 1))
+          .b_zero_point(8)
+          .bl(kr * sr * 2)
+      , test_func, isa_check));
+  gemm_tests.push_back(GemmTestParams(
+      "bl",
```

Comment:
Love this! Just for completeness can we also append `loop_k()` with a couple of steps in k_blocks.